### PR TITLE
pass index to the decorate function

### PR DESCRIPTION
**Pull request by ColinEberhardt**
By passing the index through to the decorate function it is now possible to colour the stacked bar series in a more conventional fashion:

screen shot 2015-07-06 at 20 56 41

### CODE REVIEW

**Comment by ColinEberhardt** [comment]
We'll go with this approach, but standardise on a decorate pattern as follows:

```
decorate(selection, data, index)
```

**Comment by chrisprice** [comment]
I assume that means this PR is a WIP and shouldn't be merged?

**Comment by ColinEberhardt** [comment]
> I assume that means this PR is a WIP and shouldn't be merged?

yep - update coming shortly!

**Comment by ColinEberhardt** [comment]
@chrisprice I've rolled out the agreed changes, with decorate having the following form:

    decorate(selection, data, index)

Although this change was done by find & replace and manual verification via the visual tests.

I've also updated the three 'stacked' visual tests to demonstrate the use of the `index` argument.

This is ready for review / merge.

File: site/src/components/series/stacked-bar.md [hunk 1]
```diff
@@ -58,12 +58,9 @@ example-code: |
     .xScale(x)
     .yScale(y)
     .xValue(function(d) { return d.state; })
-    .decorate(function(sel) {
-        sel.each(function(d, i) {
-            d3.select(this)
-              .selectAll('path')
-              .style({'fill': color(i), 'stroke' : color(i)});
-        });
+    .decorate(function(sel, index) {
```
Comment (chrisprice) [inline site/src/components/series/stacked-bar.md]: This looks wrong to me.

**Comment by chrisprice** [comment]
Merging despite comment, PR for comment incoming.

**Pull request merged**